dbt/include/snowflake/macros/materializations/dynamic_table/ddl.sql

```diff
@@ -15,12 +15,12 @@
 
         {%- set target_lag = configuration_changes.target_lag -%}
         {%- if target_lag -%}{{- log('Applying UPDATE TARGET_LAG to: ' ~ existing_relation) -}}{%- endif -%}
-        {%- set warehouse = configuration_changes.warehouse -%}
-        {%- if warehouse -%}{{- log('Applying UPDATE WAREHOUSE to: ' ~ existing_relation) -}}{%- endif -%}
+        {%- set snowflake_warehouse = configuration_changes.snowflake_warehouse -%}
+        {%- if snowflake_warehouse -%}{{- log('Applying UPDATE WAREHOUSE to: ' ~ existing_relation) -}}{%- endif -%}
 
         alter dynamic table {{ existing_relation }} set
             {% if target_lag %}target_lag = '{{ target_lag.context }}'{% endif %}
-            {% if warehouse %}warehouse = {{ warehouse.context }}{% endif %}
+            {% if snowflake_warehouse %}snowflake_warehouse = {{ snowflake_warehouse.context }}{% endif %}
```

Comment:
@mikealfare would we prefer to keep the variable here just as `warehouse`?

Reply:
We should keep the config variable `snowflake_warehouse`, but Snowflake expects `warehouse`. So that line should look like:

```sql
{% if snowflake_warehouse %}warehouse = {{ snowflake_warehouse.context }}{% endif %}
```

The fact that that's confusing is why I tried to make them match initially. However, users will see the error message more often than we need to maintain this (hopefully?).